### ISSUE TITLE
deprecate knife bootstrap --distro

### ISSUE THREAD

**Issue by btm** [chef/knife-windows#192]
From lib/chef/knife/bootstrap.rb:
```
      # DEPR: Remove this option in Chef 13
      option :distro,
        :short => "-d DISTRO",
        :long => "--distro DISTRO",
        :description => "Bootstrap a distro using a template. [DEPRECATED] Use -t / --bootstrap-template option instead.",
        :proc        => Proc.new { |v|
          Chef::Log.warn("[DEPRECATED] -d / --distro option is deprecated. Use -t / --bootstrap-template option instead.")
          v
        }
```

The above deprecation is from https://github.com/chef/chef/pull/1885. We should deprecate both --distro --template in knife-windows in lib/chef/knife/bootstrap_windows_base.rb.

Which is easy enough, except theres some inconsistency in --bootstrap-template in knife-windows.

again from lib/chef/knife/bootstrap.rb:
```
      option :bootstrap_template,
        :short => "-t TEMPLATE",
        :long => "--bootstrap-template TEMPLATE",
        :description => "Bootstrap Chef using a built-in or custom template. Set to the full path of an erb template or use one of the built-in templates."
```

We need to add this option to knife-windows as the preferred option as well.

**Comment by btm** [comment]
Looking at this again right now, I don't think we need --distro in knife-windows.

But we need to be consistent with the template option. Both ```knife bootstrap windows winrm``` and ```knife bootstrap windows ssh``` should have a --template-file option that matches ```knife bootstrap``` and we should deprecate --bootstrap-template and -t in favor of that (throw a warning to switch).

**Comment by siddheshwar-more** [comment]
@btm 
In chef master https://github.com/chef/chef/blob/master/lib/chef/knife/bootstrap.rb#L130-L137 `--template-file` option is deprecated and  its using `--bootstrap-template` option.

but in above comment says-  we should deprecate `--bootstrap-template` option and `-t` 
which is little misleading here, Could you please clear our confusion about which option should we deprecate for knife-windows ? 

Thank  you!

**Comment by btm** [comment]
If you look in chef master for knife bootstrap, both --distro and --template-file are deprecated, leaving --bootstrap-template/-t  as the preferred option.

In [knife-windows master](https://github.com/chef/knife-windows/blob/master/lib/chef/knife/bootstrap_windows_base.rb) we have --template-file and --distro.  To be consistent, I believe we need to add a --bootstrap-template/ -t, and to add depreciation messages to the others in knife windows.

**Comment by btm** [comment]
Fixed in #223 .
